core/modules/drr.cc

```diff
@@ -299,77 +328,34 @@ void DRR::RemoveFlow(Flow* f) {
   delete f;
 }
 
-llring* DRR::AddQueue(uint32_t slots, int* err) {
-  int bytes = llring_bytes_with_slots(slots);
-  int ret;
-
-  llring* queue = static_cast<llring*>(aligned_alloc(alignof(llring), bytes));
-  if (!queue) {
-    *err = -ENOMEM;
-    return nullptr;
-  }
-
-  ret = llring_init(queue, slots, 1, 1);
-  if (ret) {
-    std::free(queue);
-    *err = -EINVAL;
-    return nullptr;
-  }
-  return queue;
-}
 
 void DRR::Enqueue(Flow* f, bess::Packet* newpkt, int* err) {
   // if the queue is full. drop the packet.
-  if (llring_count(f->queue) >= max_queue_size_) {
+  if (f->queue->Size() >= max_queue_size_) {
     bess::Packet::Free(newpkt);
     return;
   }
 
   // creates a new queue if there is not enough space for the new packet
   // in the old queue
-  if (llring_full(f->queue)) {
+  if (f->queue->Full()) {
     uint32_t slots =
-        RoundToPowerTwo(llring_count(f->queue) * kQueueGrowthFactor);
-    f->queue = ResizeQueue(f->queue, slots, err);
-    if (*err != 0) {
+        RoundToPowerTwo(f->queue->Size() * kQueueGrowthFactor);
+    *err = f->queue->Resize(slots);
+    if (*err) {
```

Comment:
In my current implementation of LockLessQueue and Codel, the old queue is not effected if the resize fails. is that too restrictive of an assumption to make? I can put it in `resize` method description of `queue.h`